Making DropdownButtonFormField to re-render if parent widget changes

pedromassango opened the pull request:
## Description

I'm just making DropdownButtonFormField to show new value by overrrinding the `didUpdateWidget()` function and replacing the current value with the new widget instance's value.

## Related Issues

https://github.com/flutter/flutter/issues/56898

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.

pedromassango committed:
Update dropdown.dart

jlubeck commented:
I can confirm this fixes my issue: https://github.com/flutter/flutter/issues/57006
And I'm assuming it will also fix https://github.com/flutter/flutter/issues/56898

pedromassango committed:
Added tests

Added test that assert that: DropdownButtonFormField should re-render if value param changes

shihaohong commented:
I think you might have accidentally closed the PR. A closed PR cannot be reviewed until it is reopened.

pedromassango commented:
This was moved to https://github.com/flutter/flutter/pull/57037

shihaohong commented:
Ahh, I think it's because this PR was opened against your own master branch. My mistake

pedromassango commented:
Yes you got me. Ahhh
